### PR TITLE
pkg/lightning: fix integration test checkpoint_engines (#950)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #950 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In br repo:
git pr https://github.com/pingcap/br/pull/955
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/br.git pr/955:release-4.0-3e8b2e424b5d
```

---

### What problem does this PR solve? 
- Only open index engine if any data engines don't finished writing to avoid reopen closed engine in importer backend.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release Note

 - No release note

### CODE REVIEW

File: pkg/lightning/restore/restore.go [hunk 1]
```diff
@@ -1286,21 +1286,26 @@ func (t *TableRestore) restoreEngines(pCtx context.Context, rc *RestoreControlle
...
-
-		// The table checkpoint status less than `CheckpointStatusIndexImported` implies
-		// that index engine checkpoint status less than `CheckpointStatusImported`.
-		// So the index engine must be found in above process
-		if indexEngine == nil {
-			return errors.Errorf("table checkpoint status %v incompitable with index engine checkpoint status %v",
-				cp.Status, indexEngineCp.Status)
+		// import backend can't reopen engine if engine is closed, so
+		// only open index engine if any data engines don't finish writing.
+		var indexEngine *backend.OpenedEngine
```
Comment (kennytm) [inline pkg/lightning/restore/restore.go:1296]:
```
[2021-03-29T02:48:39.327Z] pkg/lightning/restore/restore.go:1296:20: undefined: backend
```

**Review by 3pointer — COMMENTED** [review]
/lgtm

**Pull request merged**